### PR TITLE
Task6

### PR DESCRIPTION
**Pull request by pakhuta**
**Steps to run:**
1. npm install
2. npm run postgres
3. npm run migrate
4. npm start

### CODE REVIEW

File: src/models/product.js [hunk 1]
```diff
@@ -1,7 +1,16 @@
-export default class Product {
-    constructor(params) {
-        this.id = params.id || process.hrtime().join('');
-        this.name = params.name || '';
-        this.reviews = params.reviews || [];
-    }
+export default function (sequelize, DataTypes) {
```
Comment (boccob) [inline src/models/product.js]: Hmm, I've seen it somewhere... Probably in `models/product.js`.

File: data/users.json [hunk 2]
```diff
@@ -1,18 +1,26 @@
 [
   {
     "id": "1",
-    "name": "Vasya"
+    "name": "vasya",
+    "password": "7815696ecbf1c96e6894b779456d330e",
```
Comment (boccob) [inline data/users.json]: Not the best way to store passwords in db.

File: src/routes/users/getUsers.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import Storage from '../../services/storage';
+
+export default async function getUsers(req, res) {
+    res.send(await Storage.get('users'));
```
Comment (boccob) [inline src/routes/users/getUsers.js]: Well, it's not a very good idea to expose sensitive data like passwords.

File: src/routes/products/getProduct.js [hunk 4]
```diff
@@ -0,0 +1,6 @@
+import Storage from '../../services/storage';
+
+export default async function getProduct(req, res) {
+    let product = await Storage.get('products', [{ name: 'id', value: req.params.id }]) || {};
```
Comment (boccob) [inline src/routes/products/getProduct.js]: Would be good to handle 404

File: src/services/storage/index.js [hunk 5]
```diff
@@ -1,70 +1,38 @@
...
     }
 
-    static add(sourceName, data) {
-        if (!storageInstance) {
-            storageInstance = new StorageInstance();
-        }
-
-        return storageInstance.add(StorageInstance.getSourceParams(sourceName), data);
+    static async add(sourceName, data) {
+        await sources[sourceName].create(data);
```
Comment (boccob) [inline src/services/storage/index.js]: Do you think that it will always succeed?

File: data/users.json (hunk 2 above)
Comment (pakhuta) [reply data/users.json]: Hmm... But where then?
From HW6 description: 8. Modify application to respond all routes from Homework 4 and return data from the database.

Comment (boccob) [reply data/users.json]: I mean it's not secure. Better to store hash and salt.

**Pull request merged**